zilliqa/src/api/eth.rs

```diff
@@ -264,8 +265,8 @@ fn get_block_transaction_count_by_number(
 
     let node = node.lock().unwrap();
     let block = match block_number {
-        BlockNumber::Number(number) => node.get_block_by_view(number),
-        BlockNumber::Earliest => node.get_block_by_view(0),
+        BlockNumber::Number(number) => node.get_block_by_number(number),
```

Comment:
I would argue we could get rid of getting blocks by view network call, and just use number, where if there are forks you could get different results  for multiple queries of number.